src/common/res/features/act-on-change/main.js

```diff
@@ -85,6 +42,72 @@
         if (ynabToolKit.debugNodes) {
           console.log('###')
         }
+        
+        // Loop through the array and act if we find a relevant changed node in the digest
+        for ( var i = 0; i < ynabToolKit.digest.length; i++ ) { 
+
+        	// Changes are detected in the category balances
+        	if ($(ynabToolKit.digest[i]).hasClass("budget-table-cell-available-div")) {
+        	  
+        	  if ( ynabToolKit.featureOptions.updateInspectorColours ) {  
+        	      ynabToolKit.updateInspectorColours();
+        	  }
+
+        	  break;
+        	  
+        	} 
+        }
+
+        for ( var i = 0; i < ynabToolKit.digest.length; i++ ) { 
+
+        	// The user has returned back to the budget screen
+        	if ($(ynabToolKit.digest[i]).hasClass('budget-inspector')) {
+        	  
+        		if ( ynabToolKit.featureOptions.checkCreditBalances ){
```

Comment:
@iamnafets yea, I understand where you are coming from. We're working through a solution so devs won't feel obliged to hack away at this single file while exposing the benefits of the mutationObserver to any enabled feature. That iteration may or may not address the feature flag checks, though.